src/config/utils.ts

```diff
@@ -133,25 +133,60 @@ export function getBoundaryLayers(): BoundaryLayerProps[] {
   const boundaryLayers = Object.values(LayerDefinitions).filter(
     (layer): layer is BoundaryLayerProps => layer.type === 'boundary',
   );
-  if (boundaryLayers.length === 0) {
-    throw new Error(
-      'No boundary layer found. There should be at least one boundary layer defined in layers.json',
-    );
-  }
+
   return boundaryLayers;
 }
 
-export function getBoundaryLayerSingleton(): BoundaryLayerProps {
+export function getDisplayBoundaryLayers(): BoundaryLayerProps[] {
   const boundaryLayers = getBoundaryLayers();
-  const boundaryLayer = boundaryLayers.find(l => l.id === 'admin_boundaries');
+  const boundariesCount = boundaryLayers.length;
 
-  if (!boundaryLayer) {
+  if (boundariesCount === 0) {
     throw new Error(
-      'No admin_boundaries Layer found! There should be exactly one boundary layer defined in layers.json with id admin_boundaries.',
+      'No boundary layer found. There should be at least one boundary layer defined in layers.json',
     );
   }
 
-  return boundaryLayer;
+  // check how many boundary layers defined in `layers.json`
+  // if they are more than one, use overrides defined in `prism.json`
+  if (boundariesCount > 1) {
+    const adminOverrideNames: LayerKey[] = get(
+      appConfig,
+      'adminBoundaryDisplayOverrides',
+      [],
+    );
+
+    const invalidOverrides = adminOverrideNames.filter(
+      id => !boundaryLayers.map(l => l.id).includes(id),
+    );
+
+    if (invalidOverrides.length > 0) {
+      throw new Error(
+        'Some of `adminBoundaryDisplayOverrides` layer Ids are not valid. You must provide valid ids from `layers.json`',
+      );
+    }
+
+    // get override layers from override names without
```

Comment:
This is actually not what we want I think.
We need to differentiate between the displayOverride (for styling) and the boundary layer that we use to get admin boundary names by default.

By construction, the union of the displayOverrides should be the same as the defaultAdminBoundary. But this is not enforced, so it is important to treat them as two different entities.